### ISSUE TITLE
No Cuda toolset found When cmake configuration with BUILD_CUDA_MODULE

### ISSUE THREAD

**Issue by Asher-1** [isl-org/Open3D#3837]
![%7(D M2Q{OT7{P0U$PJ26CM](https://user-images.githubusercontent.com/16451440/127724800-e1d49ca0-eb14-4884-8d2d-33759cadd97f.png)
**Environment (please complete the following information):**

- Operating system: Windows 10 64-bit
- Python version: Python 3.6
- Open3D version: 0.13.0
- Is this remote workstation?: no
- How did you install Open3D?:  build from source
- Compiler version VS2019
- CUDA10.1

**Additional context**

**Comment by Asher-1** [comment]
Cmake configuration when annotating enable_language(CUDA) in Cmakelist.txt  as follows:
Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19043.
Open3D 0.13.0+c31e79923
Using security hardening compiler flags: /sdl;/GS;/guard:cf and
linker flags: /INCREMENTAL:NO;/DYNAMICBASE;/HIGHENTROPYVA
CMAKE_CUDA_ARCHITECTURES: 30-real;50-real;60-real;70-real;72-real;75
Using installed third-party library 3rdparty_threads 
Using installed third-party library 3rdparty_openmp 
Building with OpenMP
Building library 3rdparty_glfw from source
Using Win32 for window creation
Building third-party library JPEG from source
NASM assembler enabled
libturbojpeg: WITH_CRT_DLL=OFF
pybind11 v2.6.2 
git version: v0.13.0-86-gc31e7992-dirty normalized to 0.13.0.86
Version: 0.13.0.86
Performing Test HAVE_STD_REGEX -- success
Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
Performing Test HAVE_POSIX_REGEX -- failed to compile
Performing Test HAVE_STEADY_CLOCK -- success
Using installed third-party library 3rdparty_opengl 
Using prebuilt third-party library Filament
Filament is located at E:/pcl_projects/Open3D/build/filament/src/ext_filament
Using MKL to support BLAS and LAPACK functionalities.
Faiss is not supported on Windows
Using installed third-party library 3rdparty_cuda_npp 10.1.243
IPP-ICV 2020.0.0 Gold available. Building interface wrappers IPP-IW.
Building PyTorch ops with CUDA
Getting PyTorch properties ...
Caffe2: CUDA detected: 10.1
Caffe2: CUDA nvcc is: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/bin/nvcc.exe
Caffe2: CUDA toolkit directory: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1
Caffe2: Header version is: 10.1
Found cuDNN: v7.6.5  (include: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include, library: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/lib/x64/cudnn.lib)
Autodetected CUDA architecture(s):  6.1
Added CUDA NVCC flags for: -gencode;arch=compute_61,code=sm_61
PyTorch         version: 1.6.0+cu101
               root dir: D:\develop\tools\Anaconda3\envs\open3d\lib\site-packages\torch
          compile flags: 
          use cxx11 abi: False
           include dirs: D:/develop/tools/Anaconda3/envs/open3d/Lib/site-packages/torch/include
           include dirs: D:/develop/tools/Anaconda3/envs/open3d/Lib/site-packages/torch/include/torch/csrc/api/include
           include dirs: C:/Program Files/NVIDIA Corporation/NvToolsExt/include
              libraries: torch
              libraries: torch_library
              libraries: D:/develop/tools/Anaconda3/envs/open3d/Lib/site-packages/torch/lib/c10.lib
              libraries: C:/Program Files/NVIDIA Corporation/NvToolsExt/lib/x64/nvToolsExt64_1.lib
              libraries: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/lib/x64/cudart_static.lib
              libraries: D:/develop/tools/Anaconda3/envs/open3d/Lib/site-packages/torch/lib/caffe2_nvrtc.lib
              libraries: D:/develop/tools/Anaconda3/envs/open3d/Lib/site-packages/torch/lib/c10_cuda.lib
PyTorch matches Open3D ABI: False == OFF
Copying E:/pcl_projects/Open3D/cpp/open3d/visualization/webrtc_server/html to E:/pcl_projects/Open3D/build/bin/resources.
PYPI_PACKAGE_NAME: open3d
================================================================================
Open3D 0.13.0 Configuration Summary
================================================================================
Enabled Features:
  OpenMP .................................. ON
  Headless Rendering ...................... OFF
  Azure Kinect Support .................... ON
  Intel RealSense Support ................. ON
  CUDA Support ............................ ON
  Build GUI ............................... ON
  Build WebRTC visualizer ................. ON
  Build Shared Library .................... OFF
  Use Windows Static Runtime .............. ON
  Build Unit Tests ........................ OFF
  Build Examples .......................... ON
  Build Python Module ..................... ON
  Build Jupyter Extension ................. ON
  Build TensorFlow Ops .................... OFF
  Build PyTorch Ops ....................... ON
  Build Benchmarks ........................ ON
  Bundle Open3D-ML ........................ ON
  Force GLIBCXX_USE_CXX11_ABI= ............ 0
================================================================================
Third-Party Dependencies:
  Eigen3 .................................. yes (build from source)
  faiss ................................... no
  filament ................................ yes (build from source)
  fmt ..................................... yes (build from source)
  GLEW .................................... yes (build from source)
  GLFW .................................... yes (build from source)
  imgui ................................... yes (build from source)
  ippicv .................................. yes (build from source)
  JPEG .................................... yes (build from source)
  jsoncpp ................................. yes (build from source)
  liblzf .................................. yes (build from source)
  OpenGL .................................. yes (build from source)
  PNG ..................................... yes (build from source)
  qhullcpp ................................ yes (build from source)
  librealsense ............................ yes (build from source)
  tinyfiledialogs ......................... yes (build from source)
  TinyGLTF ................................ yes (build from source)
  tinyobjloader ........................... yes (build from source)
  WebRTC .................................. yes (build from source)
================================================================================
Configuring done

**Comment by yxlao** [comment]
Looks like it is already working in your second post?

**Comment by Asher-1** [comment]
Only annotating 'enable_language(CUDA)' in Cmakelist.txt works , but compiling errors!

**Comment by cdrose** [comment]
@Asher-1 I am also having problems building with CUDA for Windows (see #3827). Have you made any progress?

**Comment by Asher-1** [comment]
nope



发自 网易邮箱大师




---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2021年08月09日 08:07 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [isl-org/Open3D] Open3d still do not support CUDA on windows? No Cuda toolset found When cmake configuration with BUILD_CUDA_MODULE? (#3837) |

@Asher-1 I am also having problems building with CUDA for Windows (see #3827). Have you made any progress?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ssheorey** [comment]
Hi @Asher-1 Open3D v0.14.1 has updated CMake scripts to detect CUDA toolkits. Please test with the latest release and let us know if you still face issues. Also in case of compiler errors, please post them here.

**Comment by guyuezuntinggithub** [comment]
I test it with open3d 0.16.0. Problem is still there with same info!

**Comment by bva99** [comment]
Tested with Open3D 0.19.0; Windows 11 (version 10.0.26100); Python 3.11; Visual Studio 2022, with MSVC 19.42.34432 for x64; CUDA Tookit 12.8.1.
It works here.
Although I get errors with the numeric limits and the definition of INFINITY.
